app/javascript/components/Editor.tsx

```diff
@@ -154,6 +88,11 @@ export function Editor({
 }) {
   const [tab, switchToTab] = useState(TabIndex.INSTRUCTIONS)
   const [theme, setTheme] = useState(Themes.LIGHT)
+  const [
```

Comment:
I actually think that the reducer for the revert and submission status shouldn't be here and should be abstracted behind a custom hook. I've held off on doing that for now, and will do it in a separate PR.

Reply:
I also decided to break the state up into two reducers: one that deals with the submission lifecycle, another that deals with the revert lifecycle.

Reply:
Extracting to a custom hook sounds good!